consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validation/FutureRoundProposalMessageValidator.java

```diff
@@ -15,10 +15,37 @@
 package org.hyperledger.besu.consensus.qbft.validation;
 
 import org.hyperledger.besu.consensus.qbft.messagewrappers.Proposal;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class FutureRoundProposalMessageValidator {
 
-  public boolean validateProposalMessage(final Proposal proposal) {
-    return true;
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final MessageValidatorFactory messageValidatorFactory;
+  private final long chainHeight;
+  private final BlockHeader parentHeader;
+
+  public FutureRoundProposalMessageValidator(
+      final MessageValidatorFactory messageValidatorFactory,
+      final long chainHeight,
+      final BlockHeader parentHeader) {
+    this.messageValidatorFactory = messageValidatorFactory;
+    this.chainHeight = chainHeight;
+    this.parentHeader = parentHeader;
+  }
+
+  public boolean validateProposalMessage(final Proposal msg) {
+    if (msg.getRoundIdentifier().getSequenceNumber() != chainHeight) {
```

Comment:
Doesn't this validation happen in the proposalValidator or proposalPayloadValidator? Seems out of place at this level.

Reply:
yup - it can be removed, its there as a belt-and-braces approach, but really isn't needed.